new sections: 5.4 'Creating packages' and 5.5 'code testing protocols'

jrscott opened the pull request:
still todo: how code contributors should use testreport prior to PRs

## What changes does this PR introduce?
(Bug fix, feature, docs update, ...)
docs

edoddridge commented on doc/contributing/contributing.rst:
```diff
@@ -603,29 +1096,6 @@ An example of a subroutine in/out admonition box in the documentation is :ref:`h
 An example of a calling tree in the documentation is :ref:`here <call-tree-press-meth>`.
 
-To show text from a separate file (e.g., to show lines of code, show comments from a Fortran file, show a parameter file etc.), 
-use the ``literalinclude`` directive. Example usage is shown here:
```
The content from #70 (literal include example) has been accidentally removed in this PR.

jrscott commented:
what needs to be done here to remove 'work in progress' label:

@jm-c and @christophernhill should review section 5.4 "creating packages". I had to update old text quite a bit. I have used mypackage before, but hardly an expert, so more knowledgable person to review needed.

@jm-c should review the documenting of the testreport utility and its options, how to read output etc.

@jrscott needs to figure out how to use tst 2+2 so as to document, with help from @jm-c 

@jm-c and @jrscott should clarify/write the section on what is expected in terms of contributors running testreport before submitting a PR

Need some specifics/documentation of "a test case must be provided for new code features"
albeit we did not come to agreement about this in meeting 7/18?

edoddridge commented:
Sure thing. I'll do that today.

jrscott commented:
@christophernhill  and @jm-c  please review section 5.4, extracted and updated from old 'Developer How-To" guide. There is minor overlap with 8.1, but will leave this for another day.

christophernhill commented:
@christophernhill and @jm-c to look at these.

jrscott commented:
@cnh @jahn @jm-c @edoddridge please check out section 5.5.4 in particular for discussion at weekly meeting

jrscott commented:
@edoddridge plz check 5.6.3 (citations), 5.2.1 and 5.2.2 step 7 (about tag index), 5.2.2 about git branches (after cloud figure)

christophernhill commented:
@jrscott I took a look at 5.4 and 5.5.4. Both look great and pretty much good enough to merge - pending input from @jm-c and @edoddridge. 
A small near term change would be good, to reference forward to Figure 8.1 and to chapter 8 and chapters 8, 9 and 10 in 5.4 intro. Eventually Fig 8.1 also needs updating I suspect.

jrscott commented:
Yes, I'll augment the into to 5.4 with those references/links to the figure and package chapters.
Let me look further into updating fig 8.1, how big a task that would be...

edoddridge commented:
LGTM

christophernhill commented:
@jm-c comments to be incorporated.
